pkg/apis/pipeline/v1alpha1/taskrun_validation.go

```diff
@@ -38,9 +38,15 @@ func (ts *TaskRunSpec) Validate() *apis.FieldError {
 		return apis.ErrMissingField("spec")
 	}
 
-	// Check for TaskRef
-	if ts.TaskRef.Name == "" {
-		return apis.ErrMissingField("spec.taskref.name")
+	// can't have both taskRef and taskSpec at the same time
+	if (ts.TaskRef != nil && ts.TaskRef.Name != "") && ts.TaskSpec != nil {
+		return apis.ErrDisallowedFields("spec.taskSpec")
```

Comment:
If this field takes multiple values then consider adding `spec.taskRef` along with `spec.taskSpec`

Reply:
added